pkg/tfgen/internal/unrec/example_schema_test.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2016-2024, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package unrec
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"encoding/json"
+	pschema "github.com/pulumi/pulumi/pkg/v3/codegen/schema"
+)
+
+//go:generate go run gen.go
```

Comment:
This feels like a complicated way to do this for a test. Could we get the same effect by implementing `TestMain`?

Reply:
This avoids build-depending on tfgen. Later on tfgen can use this package then. Avoid the issue with recursive packages. If that's not going to be an issue I can clean up.

Reply:
That's reasonable. Then a comment explaining ☝️ would be great.